### PR TITLE
[BUG] Complex type decimal support

### PR DESCRIPTION
**Pull request by ravikiran23**
Decimal type was failing in the complex type. 
Parsing of the decimal data type was wrong , corrected the same.

### CODE REVIEW

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala [hunk 1]
```diff
@@ -1147,30 +1162,55 @@ class CarbonSqlParser()
...
-    field.dataType.getOrElse("NIL") match {
+    val dataType = field.dataType.getOrElse("NIL")
+    dataType match {
       case "string" => Field(field.column, Some("String"), field.name, Some(null), field.parent,
-        field.storeType)
+        field.storeType
+      )
       case "integer" | "int" => Field(field.column, Some("Integer"), field.name, Some(null),
-        field.parent, field.storeType)
+        field.parent, field.storeType
```
Comment (sujith71955) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: format is not proper, displaying non modified code

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala [hunk 2]
```diff
@@ -300,7 +300,7 @@ class CarbonSqlParser()
     m.find()
     val matchedString: String = m.group(1)
     val scaleAndPrecision = matchedString.split(",")
-    (Integer.parseInt(scaleAndPrecision(0)), Integer.parseInt(scaleAndPrecision(1)))
+    (Integer.parseInt(scaleAndPrecision(0).trim), Integer.parseInt(scaleAndPrecision(1).trim))
```
Comment (sujith71955) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: Add test cases, if already present please mention the test class

Comment (ravikiran23) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: added

**Comment by sujith71955** [comment]
LGTM

**Pull request merged**